// survey-creator-angular/esm2015/toolbox/toolbox-list.component.js
import { Component, Input } from "@angular/core";
import { AngularComponentFactory, BaseAngular } from "survey-angular-ui";
import * as i0 from "@angular/core";
import * as i1 from "./toolbox-tool.component";
import * as i2 from "@angular/common";
export class ToolboxListComponent extends BaseAngular {
    getModel() {
        return this.model;
    }
}
ToolboxListComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: ToolboxListComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
ToolboxListComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: ToolboxListComponent, selector: "svc-toolbox-list ", inputs: { model: "model", creator: "creator" }, usesInheritance: true, ngImport: i0, template: "<ng-template #template>\n  <div [class]=\"model.cssClasses.root\">\n    <ng-container *ngFor=\"let item of model.renderedActions\">\n      <svc-toolbox-tool [item]=\"item\" [creator]=\"creator\" [parentModel]=\"model\" [isCompact]=\"false\"></svc-toolbox-tool>\n    </ng-container>\n  </div>\n</ng-template>", styles: [":host { display: none; }"], components: [{ type: i1.ToolboxToolComponent, selector: "svc-toolbox-tool", inputs: ["creator", "item", "parentModel", "isCompact"] }], directives: [{ type: i2.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: ToolboxListComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "svc-toolbox-list ",
                    templateUrl: "./toolbox-list.component.html",
                    styles: [":host { display: none; }"]
                }]
        }], propDecorators: { model: [{
                type: Input
            }], creator: [{
                type: Input
            }] } });
AngularComponentFactory.Instance.registerComponent("svc-toolbox-list", ToolboxListComponent);
