// survey-angular-ui/esm2015/components/popup/popup.service.js
import { Injectable, VERSION } from "@angular/core";
import { ComponentPortal, DomPortalOutlet } from "@angular/cdk/portal";
import { PopupBaseContainerComponent } from "./popup-container.component";
import * as i0 from "@angular/core";
export class PopupService {
    constructor(injector, applicationRef, componentFactoryResolver) {
        this.injector = injector;
        this.applicationRef = applicationRef;
        this.componentFactoryResolver = componentFactoryResolver;
    }
    createComponent(popupViewModel) {
        let portalHost;
        if (Number.parseInt(VERSION.major) > 19) {
            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
            //@ts-ignore
            portalHost = new DomPortalOutlet(popupViewModel.container, this.applicationRef, this.injector);
        }
        else {
            portalHost = new DomPortalOutlet(popupViewModel.container, this.componentFactoryResolver, this.applicationRef, this.injector);
        }
        const portal = new ComponentPortal(PopupBaseContainerComponent);
        const componentRef = portalHost.attach(portal);
        popupViewModel.setComponentElement(popupViewModel.container.children[0]);
        componentRef.instance.model = popupViewModel;
        componentRef.changeDetectorRef.detectChanges();
        return portalHost;
    }
}
PopupService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: PopupService, deps: [{ token: i0.Injector }, { token: i0.ApplicationRef }, { token: i0.ComponentFactoryResolver }], target: i0.ɵɵFactoryTarget.Injectable });
PopupService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: PopupService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: PopupService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i0.Injector }, { type: i0.ApplicationRef }, { type: i0.ComponentFactoryResolver }]; } });
