// survey-creator-angular/esm2015/components/action-button.component.js
import { Component, Input } from "@angular/core";
import { AngularComponentFactory, EmbeddedViewContentComponent } from "survey-angular-ui";
import * as i0 from "@angular/core";
import * as i1 from "survey-angular-ui";
import * as i2 from "@angular/common";
export class ActionButtonComponent extends EmbeddedViewContentComponent {
    constructor() {
        super(...arguments);
        this.classes = "";
        this.selected = false;
        this.disabled = false;
        this.text = "";
        this.title = "";
        this.iconName = "";
        this.allowBubble = false;
    }
    onClick(event) {
        this.click();
        if (!this.allowBubble) {
            event.stopPropagation();
        }
    }
}
ActionButtonComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: ActionButtonComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
ActionButtonComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: ActionButtonComponent, selector: "svc-action-button", inputs: { classes: "classes", click: "click", selected: "selected", disabled: "disabled", text: "text", title: "title", iconName: "iconName", allowBubble: "allowBubble" }, usesInheritance: true, ngImport: i0, template: "<ng-template #template>\n  <ng-container *ngIf=\"!iconName\">\n    <span *ngIf=\"disabled\" class=\"svc-action-button svc-action-button--disabled\" [class]=\"classes\"\n      [attr.title]=\"title\">{{text}}</span>\n    <span *ngIf=\"!disabled\" role=\"button\" class=\"svc-action-button\" (click)=\"onClick($event)\" [key2click]\n      [attr.title]=\"title\" [class.svc-action-button--selected]=\"selected\" [class]=\"classes\">{{text}}</span>\n  </ng-container>\n  <ng-container *ngIf=\"iconName\">\n    <span *ngIf=\"disabled\" class=\"svc-action-button svc-action-button--disabled svc-action-button--icon\"\n      [class]=\"classes\" [attr.title]=\"title\">\n      <svg [iconName]=\"iconName\" [size]=\"'auto'\" sv-ng-svg-icon></svg>\n    </span>\n    <span *ngIf=\"!disabled\" class=\"svc-action-button svc-action-button--icon\" (click)=\"onClick($event)\" [key2click]\n      [attr.title]=\"title\" [attr.aria-label]=\"text\" [class]=\"classes\" [class.svc-action-button--selected]=\"selected\">\n      <svg [iconName]=\"iconName\" [size]=\"'auto'\" sv-ng-svg-icon></svg>\n    </span>\n  </ng-container>\n</ng-template>", styles: [":host { display: none; }"], components: [{ type: i1.SvgIconComponent, selector: "'[sv-ng-svg-icon]'", inputs: ["size", "width", "height", "iconName", "partCss", "css", "title"] }], directives: [{ type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i1.Key2ClickDirective, selector: "[key2click]", inputs: ["key2click"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: ActionButtonComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "svc-action-button",
                    templateUrl: "./action-button.component.html",
                    styles: [":host { display: none; }"]
                }]
        }], propDecorators: { classes: [{
                type: Input
            }], click: [{
                type: Input
            }], selected: [{
                type: Input
            }], disabled: [{
                type: Input
            }], text: [{
                type: Input
            }], title: [{
                type: Input
            }], iconName: [{
                type: Input
            }], allowBubble: [{
                type: Input
            }] } });
AngularComponentFactory.Instance.registerComponent("svc-action-button", ActionButtonComponent);
