// survey-creator-angular/esm2015/components/scroll.component.js
import { ScrollViewModel } from "survey-creator-core";
import { Component, ViewChild } from "@angular/core";
import { AngularComponentFactory, EmbeddedViewContentComponent } from "survey-angular-ui";
import * as i0 from "@angular/core";
export class ScrollComponent extends EmbeddedViewContentComponent {
    constructor(viewContainerRef) {
        super(viewContainerRef);
        this.model = new ScrollViewModel();
    }
    ngAfterViewInit() {
        var _a;
        this.model.setRootElement((_a = this.container) === null || _a === void 0 ? void 0 : _a.nativeElement);
    }
    ngOnDestroy() {
        this.model.setRootElement(undefined);
        this.model.unsubscribeRootElement();
    }
}
ScrollComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: ScrollComponent, deps: [{ token: i0.ViewContainerRef }], target: i0.ɵɵFactoryTarget.Component });
ScrollComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: ScrollComponent, selector: "svc-scroll", viewQueries: [{ propertyName: "container", first: true, predicate: ["container"], descendants: true }], usesInheritance: true, ngImport: i0, template: "<ng-template #template>\n    <div class=\"svc-scroll__wrapper\" #container>\n        <div class=\"svc-scroll__scroller sv-drag-target-skipped\" (scroll)=\"model.onScrollContainer()\">\n            <div class=\"svc-scroll__container\">\n                <ng-content></ng-content>\n            </div>\n        </div>\n        <div class=\"svc-scroll__scrollbar\" (scroll)=\"model.onScrollScrollbar()\">\n            <div class=\"svc-scroll__scrollbar-sizer\">\n            </div>\n        </div>\n    </div>\n</ng-template>", styles: [":host { display: none; }"] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: ScrollComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "svc-scroll",
                    templateUrl: "./scroll.component.html",
                    styles: [":host { display: none; }"]
                }]
        }], ctorParameters: function () { return [{ type: i0.ViewContainerRef }]; }, propDecorators: { container: [{
                type: ViewChild,
                args: ["container"]
            }] } });
AngularComponentFactory.Instance.registerComponent("svc-scroll", ScrollComponent);
