// survey-angular-ui/esm2015/components/scroll/scroll.component.js
import { ScrollViewModel } from "survey-core";
import { Component, ViewChild, Input } from "@angular/core";
import { EmbeddedViewContentComponent } from "../../embedded-view-content.component";
import { AngularComponentFactory } from "./../../component-factory";
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
export class ScrollComponent extends EmbeddedViewContentComponent {
    constructor(viewContainerRef) {
        super(viewContainerRef);
        this.model = new ScrollViewModel();
        this.model.onInnerHeightChanged = (hasScroll) => {
            if (!!this.onInnerHeightChanged) {
                this.onInnerHeightChanged(hasScroll);
            }
        };
    }
    ngAfterViewInit() {
        var _a;
        this.model.setRootElement((_a = this.container) === null || _a === void 0 ? void 0 : _a.nativeElement);
    }
    ngOnDestroy() {
        this.model.setRootElement(undefined);
        this.model.unsubscribeRootElement();
    }
}
ScrollComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: ScrollComponent, deps: [{ token: i0.ViewContainerRef }], target: i0.ɵɵFactoryTarget.Component });
ScrollComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: ScrollComponent, selector: "sv-scroll", inputs: { disabled: "disabled", onInnerHeightChanged: "onInnerHeightChanged" }, viewQueries: [{ propertyName: "container", first: true, predicate: ["container"], descendants: true }], usesInheritance: true, ngImport: i0, template: "<ng-template #template>\n    <ng-container *ngIf=\"!disabled\">\n        <div class=\"sv-scroll__wrapper\" #container>\n            <div class=\"sv-scroll__scroller sv-drag-target-skipped\" (scroll)=\"model.onScrollContainer()\">\n                <div class=\"sv-scroll__container\">\n                    <ng-container *ngTemplateOutlet=\"transparentTemplate\"></ng-container>\n                </div>\n            </div>\n            <div class=\"sv-scroll__scrollbar\" (scroll)=\"model.onScrollScrollbar()\">\n                <div class=\"sv-scroll__scrollbar-sizer\">\n                </div>\n            </div>\n        </div>\n    </ng-container>\n    <ng-container *ngIf=\"disabled\"><ng-container *ngTemplateOutlet=\"transparentTemplate\"></ng-container></ng-container>\n</ng-template>\n<ng-template #transparentTemplate>\n    <ng-content></ng-content>\n</ng-template>", styles: [":host { display: none; }"], directives: [{ type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i1.NgTemplateOutlet, selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: ScrollComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "sv-scroll",
                    templateUrl: "./scroll.component.html",
                    styles: [":host { display: none; }"]
                }]
        }], ctorParameters: function () { return [{ type: i0.ViewContainerRef }]; }, propDecorators: { disabled: [{
                type: Input
            }], onInnerHeightChanged: [{
                type: Input
            }], container: [{
                type: ViewChild,
                args: ["container"]
            }] } });
AngularComponentFactory.Instance.registerComponent("sv-scroll", ScrollComponent);
