// survey-angular-ui/esm2015/base-angular.js
import { Component } from "@angular/core";
import { EmbeddedViewContentComponent } from "./embedded-view-content.component";
import * as i0 from "@angular/core";
export class BaseAngular extends EmbeddedViewContentComponent {
    constructor(changeDetectorRef, viewContainerRef) {
        super(viewContainerRef);
        this.changeDetectorRef = changeDetectorRef;
        this.isModelSubsribed = false;
        this.isDestroyed = false;
        this.onArrayChangedCallback = (stateElement, options) => {
            this.update(options.name);
        };
        this.onPropertyChangedCallback = (stateElement, options) => {
            this.update(options.name);
        };
        this.isUpdatesBlocked = false;
    }
    get surveyModel() {
        return this.getModel().getSurvey();
    }
    ngDoCheck() {
        super.ngDoCheck();
        if (this.previousModel !== this.getModel()) {
            this.unMakeBaseElementAngular(this.previousModel);
            this.makeBaseElementAngular(this.getModel());
            this.onModelChanged();
            this.previousModel = this.getModel();
        }
        this.setIsModelRendering(true);
    }
    onModelChanged() { }
    setIsModelRendering(val) {
        const model = this.getModel();
        if (!!model) {
            model.isRendering = val;
        }
    }
    getIsModelRendering(stateElement) {
        const model = stateElement !== null && stateElement !== void 0 ? stateElement : this.getModel();
        return !!model && !!model.isRendering;
    }
    ngOnDestroy() {
        this.isDestroyed = true;
        this.unMakeBaseElementAngular(this.getModel());
        this.previousModel = undefined;
    }
    isBaseElementSubsribed(stateElement) {
        var _a;
        return ((_a = stateElement.__ngImplementedCount) !== null && _a !== void 0 ? _a : 0) > 0;
    }
    makeBaseElementAngular(stateElement) {
        var _a;
        if (!!stateElement && !this.getIsModelRendering(stateElement)) {
            this.isModelSubsribed = true;
            stateElement.addOnArrayChangedCallback(this.onArrayChangedCallback);
            stateElement.addOnPropertyValueChangedCallback(this.onPropertyChangedCallback);
            stateElement.enableOnElementRerenderedEvent();
            stateElement.__ngImplementedCount = ((_a = stateElement.__ngImplementedCount) !== null && _a !== void 0 ? _a : 0) + 1;
        }
    }
    unMakeBaseElementAngular(stateElement) {
        var _a;
        if (!!stateElement && this.isModelSubsribed) {
            this.isModelSubsribed = false;
            stateElement.removeOnPropertyValueChangedCallback(this.onPropertyChangedCallback);
            stateElement.removeOnArrayChangedCallback(this.onArrayChangedCallback);
            stateElement.disableOnElementRerenderedEvent();
            if (((_a = stateElement.__ngImplementedCount) !== null && _a !== void 0 ? _a : 0) - 1 <= 0) {
                delete stateElement.__ngImplementedCount;
            }
        }
    }
    update(key) {
        if (this.getIsModelRendering() || this.isUpdatesBlocked)
            return;
        if (key && this.getPropertiesToUpdateSync().indexOf(key) > -1) {
            this.beforeUpdate();
            this.detectChanges();
            this.afterUpdate(true);
        }
        else {
            this.isUpdatesBlocked = true;
            queueMicrotask(() => {
                if (!this.isDestroyed) {
                    this.isUpdatesBlocked = false;
                    this.beforeUpdate();
                    this.detectChanges();
                    this.afterUpdate();
                }
            });
        }
    }
    getChangeDetectorRef() {
        return this.embeddedView ? this.embeddedView : this.changeDetectorRef;
    }
    getPropertiesToUpdateSync() {
        return [];
    }
    detectChanges() {
        this.getChangeDetectorRef().detectChanges();
    }
    getShouldReattachChangeDetector() {
        return true;
    }
    beforeUpdate() {
        this.setIsModelRendering(true);
    }
    afterUpdate(isSync = false) {
        this.setIsModelRendering(false);
        const model = this.getModel();
        if (model && !this.isDestroyed) {
            model.afterRerender();
        }
    }
    ngAfterViewChecked() {
        super.ngAfterViewChecked();
        this.setIsModelRendering(false);
    }
}
BaseAngular.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: BaseAngular, deps: [{ token: i0.ChangeDetectorRef }, { token: i0.ViewContainerRef }], target: i0.ɵɵFactoryTarget.Component });
BaseAngular.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: BaseAngular, selector: "ng-component", usesInheritance: true, ngImport: i0, template: "", isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: BaseAngular, decorators: [{
            type: Component,
            args: [{
                    template: ""
                }]
        }], ctorParameters: function () { return [{ type: i0.ChangeDetectorRef }, { type: i0.ViewContainerRef }]; } });
