// survey-creator-angular/esm2015/components/switcher.component.js
import { Component, Input } from "@angular/core";
import { AngularComponentFactory, BaseAngular } from "survey-angular-ui";
import * as i0 from "@angular/core";
import * as i1 from "survey-angular-ui";
import * as i2 from "@angular/common";
export class SwitcherComponent extends BaseAngular {
    getModel() {
        return this.model;
    }
}
SwitcherComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: SwitcherComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
SwitcherComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: SwitcherComponent, selector: "svc-switcher", inputs: { model: "model" }, usesInheritance: true, ngImport: i0, template: "<ng-template #template>\n  <button type=\"button\" (click)=\"model.action(model, model.getIsTrusted($event))\"\n    [key2click]=\"{ processEsc: false, disableTabStop: model.disableTabStop }\" [class]=\"model.getActionBarItemCss()\"\n    [disabled]=\"model.disabled\" [attr.title]=\"model.tooltip || model.title\" [attr.aria-checked]=\"model.ariaChecked\"\n    [attr.aria-expanded]=\"model.ariaExpanded\" [attr.role]=\"model.ariaRole\">\n    <div [class]=\"model.getSwitcherIconCss()\">\n      <div class=\"svc-switcher__icon-thumb\"></div>\n    </div>\n    <span *ngIf=\"model.hasTitle\" class=\"svc-switcher__title\">{{ model.title }}</span>\n  </button>\n  <ng-template>", styles: [":host { display: none; }"], directives: [{ type: i1.Key2ClickDirective, selector: "[key2click]", inputs: ["key2click"] }, { type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: SwitcherComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "svc-switcher",
                    templateUrl: "./switcher.component.html",
                    styles: [":host { display: none; }"]
                }]
        }], propDecorators: { model: [{
                type: Input
            }] } });
AngularComponentFactory.Instance.registerComponent("svc-switcher", SwitcherComponent);
