// survey-creator-angular/esm2015/custom-questions/color-item.component.js
import { Component, Input } from "@angular/core";
import { AngularComponentFactory, BaseAngular } from "survey-angular-ui";
import * as i0 from "@angular/core";
import * as i1 from "survey-angular-ui";
export class ColorItemComponent extends BaseAngular {
    getModel() {
        return this.model;
    }
    getStyle() {
        return { backgroundColor: this.model.value };
    }
}
ColorItemComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: ColorItemComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
ColorItemComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: ColorItemComponent, selector: "svc-color-item", inputs: { model: "model" }, usesInheritance: true, ngImport: i0, template: "<ng-template #template>\n  <span class=\"spg-color-editor__color-swatch\" [style]=\"getStyle()\"></span>\n  <sv-ng-string [model]=\"model.locTitle\"></sv-ng-string>\n</ng-template>", styles: [":host { display: none }"], components: [{ type: i1.SurveyStringComponent, selector: "sv-ng-string, '[sv-ng-string]'", inputs: ["model"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: ColorItemComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "svc-color-item",
                    templateUrl: "./color-item.component.html",
                    styles: [":host { display: none }"]
                }]
        }], propDecorators: { model: [{
                type: Input
            }] } });
AngularComponentFactory.Instance.registerComponent("color-item", ColorItemComponent);
