// survey-creator-angular/esm2015/page-wrapper.component.js
import { Component, Input } from "@angular/core";
import { EmbeddedViewContentComponent } from "survey-angular-ui";
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
import * as i2 from "survey-angular-ui";
export class PageWrapperComponent extends EmbeddedViewContentComponent {
    constructor(viewContainerRef, changeDetectorRef) {
        super(viewContainerRef);
        this.changeDetectorRef = changeDetectorRef;
        this.id = `PageWrapperCallback_${++PageWrapperComponent.ID}`;
    }
    setupCallback(page) {
        if (!page)
            return;
        page.registerFunctionOnPropertyValueChanged("name", () => {
            this.embeddedView ? this.embeddedView.detectChanges() : this.changeDetectorRef.detectChanges();
        }, this.id);
    }
    clearCallback(page) {
        if (!page)
            return;
        page.unRegisterFunctionOnPropertyValueChanged("name");
    }
    ngOnChanges(changes) {
        if (changes.page && changes.page.currentValue !== changes.page.previousValue) {
            this.clearCallback(changes.page.previousValue);
            this.setupCallback(changes.page.currentValue);
        }
    }
    ngOnDestroy() {
        this.clearCallback(this.page);
    }
}
PageWrapperComponent.ID = 0;
PageWrapperComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: PageWrapperComponent, deps: [{ token: i0.ViewContainerRef }, { token: i0.ChangeDetectorRef }], target: i0.ɵɵFactoryTarget.Component });
PageWrapperComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: PageWrapperComponent, selector: "svc-page-wrapper", inputs: { page: "page", isGhost: "isGhost", creator: "creator", className: "className" }, usesInheritance: true, usesOnChanges: true, ngImport: i0, template: "<ng-template #template>\n    <div\n      [class]=\"className\"\n      [attr.data-sv-drop-target-page]=\"page?.name\"\n      [attr.data-sv-drop-target-survey-element]=\"isGhost ? 'newGhostPage' : page?.name\"\n    >\n      <ng-container *ngIf=\"!!page\">\n           <ng-template [component]=\"{ name: 'svc-page', data: { model: page, creator, survey: creator.survey, isGhost }}\"></ng-template>\n      </ng-container>\n    </div>\n</ng-template>", styles: [":host { display: none; }"], directives: [{ type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i2.DynamicComponentDirective, selector: "[component]", inputs: ["component"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: PageWrapperComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "svc-page-wrapper",
                    templateUrl: "./page-wrapper.component.html",
                    styles: [":host { display: none; }"]
                }]
        }], ctorParameters: function () { return [{ type: i0.ViewContainerRef }, { type: i0.ChangeDetectorRef }]; }, propDecorators: { page: [{
                type: Input
            }], isGhost: [{
                type: Input
            }], creator: [{
                type: Input
            }], className: [{
                type: Input
            }] } });
