// survey-creator-angular/esm2015/page-element-content.component.js
import { Component, Input } from "@angular/core";
import { EmbeddedViewContentComponent } from "survey-angular-ui";
import * as i0 from "@angular/core";
import * as i1 from "survey-angular-ui";
export class PageElementContentComponent extends EmbeddedViewContentComponent {
    ngDoCheck() { }
    ngOnChanges(changes) {
        var _a;
        (_a = this.embeddedView) === null || _a === void 0 ? void 0 : _a.reattach();
    }
}
PageElementContentComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: PageElementContentComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
PageElementContentComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: PageElementContentComponent, selector: "svc-page-element-content", inputs: { model: "model", survey: "survey" }, usesInheritance: true, usesOnChanges: true, ngImport: i0, template: "<ng-template #template><page [model]=\"model\" [survey]=\"survey\"></page></ng-template>", isInline: true, styles: [":host { display: none }"], components: [{ type: i1.PageComponent, selector: "sv-page, page, sv-ng-page", inputs: ["model", "survey"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: PageElementContentComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "svc-page-element-content",
                    template: "<ng-template #template><page [model]=\"model\" [survey]=\"survey\"></page></ng-template>",
                    styles: [":host { display: none }"]
                }]
        }], propDecorators: { model: [{
                type: Input
            }], survey: [{
                type: Input
            }] } });
