// survey-creator-angular/esm2015/page.component.js
import { Component, ElementRef, Input, ViewChild } from "@angular/core";
import { AngularComponentFactory } from "survey-angular-ui";
import { PageAdorner } from "survey-creator-core";
import { CreatorModelComponent } from "./creator-model.component";
import * as i0 from "@angular/core";
import * as i1 from "./page-element-content.component";
import * as i2 from "survey-angular-ui";
import * as i3 from "@angular/common";
export class PageDesignerComponent extends CreatorModelComponent {
    createModel() {
        if (this.adorner) {
            this.adorner.dispose();
        }
        if (this.model) {
            this.adorner = new PageAdorner(this.creator, this.model);
            this.adorner.isGhost = this.isGhost;
        }
    }
    ngDoCheck() {
        super.ngDoCheck();
        if (this.adorner) {
            this.adorner.isGhost = this.isGhost;
        }
    }
    getModel() {
        return this.adorner;
    }
    getPropertiesToTrack() {
        return ["creator", "model"];
    }
    ngAfterViewChecked() {
        var _a;
        super.ngAfterViewChecked();
        if (this.adorner && ((_a = this.container) === null || _a === void 0 ? void 0 : _a.nativeElement)) {
            this.adorner.setRootElement(this.container.nativeElement);
        }
    }
    ngOnDestroy() {
        super.ngOnDestroy();
        this.adorner.dispose();
    }
}
PageDesignerComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: PageDesignerComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
PageDesignerComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: PageDesignerComponent, selector: "svc-page", inputs: { model: "model", survey: "survey", creator: "creator", isGhost: "isGhost" }, viewQueries: [{ propertyName: "container", first: true, predicate: ["container"], descendants: true, read: ElementRef }], usesInheritance: true, ngImport: i0, template: "<ng-template #template>\n<div *ngIf=\"model && adorner\" class=\"svc-page__content\" [class]=\"adorner.css\" [key2click]\n  (click)=\"adorner.select(adorner, $event)\" (dblclick)=\"adorner.dblclick($event)\"\n  (mouseover)=\"adorner.hover($event, $event.currentTarget)\" (mouseleave)=\"adorner.hover($event, $event.target)\"\n  [id]=\"adorner.page.id\" [attr.data-sv-drop-target-survey-page]=\"adorner.dropTargetName\" #container>\n  <div class=\"svc-question__drop-indicator svc-question__drop-indicator--top\"></div>\n  <div class=\"svc-question__drop-indicator svc-question__drop-indicator--bottom\"></div>\n  <svc-page-element-content\n    *ngIf=\"adorner.needRenderContent\" \n    [model]=\"adorner.page\" \n    [survey]=\"survey\">\n  </svc-page-element-content>\n  <div *ngIf=\"!adorner.needRenderContent\" class=\"svc-page__loading-content\">\n    <sv-ng-loading-indicator></sv-ng-loading-indicator>\n  </div>\n  <div *ngIf=\"adorner.showPlaceholder\" class=\"svc-page__placeholder_frame\">\n    <div class=\"svc-panel__placeholder_frame\">\n      <div class=\"svc-panel__placeholder\">{{adorner.placeholderText}}</div>\n    </div>\n  </div>\n  <div *ngIf=\"adorner.allowDragging && !adorner.isGhost\" class=\"svc-question__drag-area\"\n    (pointerdown)=\"adorner.onPointerDown($event)\">\n    <svg class=\"svc-question__drag-element\" [iconName]=\"'icon-drag-area-indicator_24x16'\" [size]=\"'auto'\"\n      sv-ng-svg-icon></svg>\n    <div class=\"svc-page__content-actions\">\n      <sv-action-bar [model]=\"adorner.actionContainer\" #container></sv-action-bar>\n      <sv-action-bar *ngIf=\"adorner.topActionContainer.hasActions\" [model]=\"adorner.topActionContainer\"></sv-action-bar>\n    </div>\n  </div>\n  <div *ngIf=\"!adorner.allowDragging || adorner.isGhost\" class=\"svc-page__content-actions\">\n    <sv-action-bar [model]=\"adorner.actionContainer\"></sv-action-bar>\n    <sv-action-bar *ngIf=\"adorner.topActionContainer.hasActions\" [model]=\"adorner.topActionContainer\"></sv-action-bar>\n  </div>\n  <sv-action-bar [model]=\"adorner.footerActionsBar\"></sv-action-bar>\n</div>\n</ng-template>", styles: [":host { display: none; }"], components: [{ type: i1.PageElementContentComponent, selector: "svc-page-element-content", inputs: ["model", "survey"] }, { type: i2.LoadingIndicatorComponent, selector: "sv-ng-loading-indicator" }, { type: i2.SvgIconComponent, selector: "'[sv-ng-svg-icon]'", inputs: ["size", "width", "height", "iconName", "partCss", "css", "title"] }, { type: i2.ActionBarComponent, selector: "sv-action-bar, sv-ng-action-bar", inputs: ["model", "handleClick"] }], directives: [{ type: i3.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i2.Key2ClickDirective, selector: "[key2click]", inputs: ["key2click"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: PageDesignerComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "svc-page",
                    templateUrl: "./page.component.html",
                    styles: [":host { display: none; }"]
                }]
        }], propDecorators: { model: [{
                type: Input
            }], survey: [{
                type: Input
            }], creator: [{
                type: Input
            }], isGhost: [{
                type: Input
            }], container: [{
                type: ViewChild,
                args: ["container", { read: ElementRef }]
            }] } });
AngularComponentFactory.Instance.registerComponent("svc-page", PageDesignerComponent);
