// survey-creator-angular/esm2015/question-element-content.component.js
import { Component, Input } from "@angular/core";
import { EmbeddedViewContentComponent } from "survey-angular-ui";
import * as i0 from "@angular/core";
import * as i1 from "survey-angular-ui";
export class QuestionElementContentComponent extends EmbeddedViewContentComponent {
    ngDoCheck() { }
    ngOnChanges(changes) {
        var _a;
        (_a = this.embeddedView) === null || _a === void 0 ? void 0 : _a.reattach();
    }
}
QuestionElementContentComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: QuestionElementContentComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
QuestionElementContentComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: QuestionElementContentComponent, selector: "svc-question-element-content", inputs: { componentName: "componentName", componentData: "componentData" }, usesInheritance: true, usesOnChanges: true, ngImport: i0, template: "<ng-template #template><ng-template [component]=\"{ name: componentName, data: componentData }\"></ng-template></ng-template>", isInline: true, styles: [":host { display: none }"], directives: [{ type: i1.DynamicComponentDirective, selector: "[component]", inputs: ["component"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: QuestionElementContentComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "svc-question-element-content",
                    template: "<ng-template #template><ng-template [component]=\"{ name: componentName, data: componentData }\"></ng-template></ng-template>",
                    styles: [":host { display: none }"]
                }]
        }], propDecorators: { componentName: [{
                type: Input
            }], componentData: [{
                type: Input
            }] } });
