// survey-creator-angular/esm2015/question-editor.component.js
import { Component, Input } from "@angular/core";
import { AngularComponentFactory, EmbeddedViewContentComponent } from "survey-angular-ui";
import * as i0 from "@angular/core";
import * as i1 from "survey-angular-ui";
export class QuestionEditorComponent extends EmbeddedViewContentComponent {
    get question() {
        return this.survey.getAllQuestions()[0];
    }
    get elementComponentName() {
        return this.question.isPanel ? "panel" : "question";
    }
    get componentName() {
        const survey = this.survey;
        if (!!survey) {
            const name = survey.getElementWrapperComponentName(this.question);
            if (!!name) {
                return name;
            }
        }
        return this.elementComponentName;
    }
    get componentData() {
        const survey = this.survey;
        let data;
        if (!!survey) {
            data = survey.getElementWrapperComponentData(this.question);
        }
        return {
            componentName: this.elementComponentName,
            componentData: {
                model: this.question,
                data: data
            }
        };
    }
}
QuestionEditorComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: QuestionEditorComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
QuestionEditorComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: QuestionEditorComponent, selector: "svc-question-editor-content", inputs: { survey: "survey", style: "style" }, usesInheritance: true, ngImport: i0, template: "<ng-template #template>\n  <div [style]=\"style\">\n    <ng-template [component]=\"{ name: componentName, data: componentData }\"></ng-template>\n  </div>\n</ng-template>", directives: [{ type: i1.DynamicComponentDirective, selector: "[component]", inputs: ["component"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: QuestionEditorComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "svc-question-editor-content",
                    templateUrl: "./question-editor.component.html"
                }]
        }], propDecorators: { survey: [{
                type: Input
            }], style: [{
                type: Input
            }] } });
AngularComponentFactory.Instance.registerComponent("svc-question-editor-content", QuestionEditorComponent);
