// survey-angular-ui/esm2015/components/file/file-page.component.js
import { Component, Input } from "@angular/core";
import { BaseAngular } from "../../base-angular";
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
import * as i2 from "../../utils/dynamic.directive";
export class FilePageComponent extends BaseAngular {
    getModel() {
        return this.page;
    }
}
FilePageComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: FilePageComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
FilePageComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: FilePageComponent, selector: "sv-ng-file-page", inputs: { page: "page", question: "question" }, usesInheritance: true, ngImport: i0, template: "<ng-template #template>\n    <div [class]=\"page.css\" [attr.id]=\"page.id\">\n        <ng-container *ngFor=\"let item of page.items\">\n            <ng-template [component]=\"{ name: 'sv-file-item', data: { item, question } }\"></ng-template>\n        </ng-container>\n    </div>\n</ng-template>", styles: [":host{display:none}\n"], directives: [{ type: i1.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { type: i2.DynamicComponentDirective, selector: "[component]", inputs: ["component"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: FilePageComponent, decorators: [{
            type: Component,
            args: [{
                    templateUrl: "./file-page.component.html",
                    selector: "sv-ng-file-page",
                    styleUrls: ["../../hide-host.scss"]
                }]
        }], propDecorators: { page: [{
                type: Input
            }], question: [{
                type: Input
            }] } });
