// survey-angular-ui/esm2015/components/matrixdynamicplaceholder/matrixdynamicplaceholder.component.js
import { Component, Input } from "@angular/core";
import { AngularComponentFactory } from "../../component-factory";
import * as i0 from "@angular/core";
import * as i1 from "../../survey-string.component";
import * as i2 from "@angular/common";
export class MatrixDynamicPlaceholderComponent {
    addRowClick() {
        this.question.addRowUI();
    }
}
MatrixDynamicPlaceholderComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: MatrixDynamicPlaceholderComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
MatrixDynamicPlaceholderComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: MatrixDynamicPlaceholderComponent, selector: "sv-ng-placeholder-matrixdynamic", inputs: { question: "question" }, ngImport: i0, template: "<div [class]=\"question.cssClasses.noRowsSection\">\n  <div [class]=\"question.cssClasses.noRowsText\">\n    <sv-ng-string [model]=\"question.locNoRowsText\"></sv-ng-string>\n  </div>\n  <button\n    *ngIf=\"question.renderedTable.showAddRow\"\n    type=\"button\"\n    [class]=\"question.getAddRowButtonCss(true)\"\n    (click)=\"addRowClick()\"\n  >\n    <sv-ng-string [model]=\"question.locAddRowText\"></sv-ng-string>\n    <span [class]=\"question.cssClasses.iconAdd\"></span>\n  </button>\n</div> ", components: [{ type: i1.SurveyStringComponent, selector: "sv-ng-string, '[sv-ng-string]'", inputs: ["model"] }], directives: [{ type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: MatrixDynamicPlaceholderComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "sv-ng-placeholder-matrixdynamic",
                    templateUrl: "./matrixdynamicplaceholder.component.html"
                }]
        }], propDecorators: { question: [{
                type: Input
            }] } });
AngularComponentFactory.Instance.registerComponent("sv-ng-placeholder-matrixdynamic", MatrixDynamicPlaceholderComponent);
