// survey-creator-angular/esm2015/questions/logic-operator.component.js
import { Component } from "@angular/core";
import { AngularComponentFactory, QuestionAngular } from "survey-angular-ui";
import { DropdownListModel, RendererFactory } from "survey-core";
import { initLogicOperator } from "survey-creator-core";
import * as i0 from "@angular/core";
import * as i1 from "survey-angular-ui";
import * as i2 from "@angular/common";
export class LogicOperatorComponent extends QuestionAngular {
    get dropdownModel() {
        if (!this.dropdownListModel) {
            this.dropdownListModel = new DropdownListModel(this.model);
        }
        return this.dropdownListModel;
    }
    click(event) {
        var _a;
        (_a = this.dropdownListModel) === null || _a === void 0 ? void 0 : _a.onClick(event);
    }
    keyup(event) {
        var _a;
        (_a = this.dropdownListModel) === null || _a === void 0 ? void 0 : _a.keyHandler(event);
    }
    ngOnInit() {
        super.ngOnInit();
        this.dropdownListModel = this.model.dropdownListModel || new DropdownListModel(this.model);
        initLogicOperator(this.model);
    }
}
LogicOperatorComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: LogicOperatorComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
LogicOperatorComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: LogicOperatorComponent, selector: "svc-logic-operator", usesInheritance: true, ngImport: i0, template: "<ng-template #template>\n  <div [class]=\"model.cssClasses.selectWrapper\">\n    <ng-container *ngIf=\"!model.isReadOnly\">\n      <div [class]=\"model.getControlClass()\"  (click)=\"click($event)\" (keyup)=\"keyup($event)\"\n        [attr.id]=\"model.inputId\" \n        [attr.required]=\"model.isRequired\" \n        [attr.tabindex]= \"model.isInputReadOnly ? undefined : 0\"\n        [attr.disabled]=\"model.isInputReadOnly\"\n        [attr.role]=\"dropdownModel.ariaQuestionRole\"\n        [attr.aria-required]=\"dropdownModel.ariaQuestionRequired\"\n        [attr.aria-invalid]=\"dropdownModel.ariaQuestionInvalid\"\n        [attr.aria-errormessage]=\"dropdownModel.ariaQuestionErrorMessage\" \n        [attr.aria-expanded]=\"dropdownModel.ariaQuestionExpanded\"\n        [attr.aria-label]=\"dropdownModel.ariaQuestionLabel\" \n        [attr.aria-labelledby]=\"dropdownModel.ariaQuestionLabelledby\"\n        [attr.aria-controls]=\"dropdownModel.ariaQuestionControls\"\n      >\n        <div [class]=\"model.cssClasses.controlValue\">\n          <sv-ng-string *ngIf=\"model.locReadOnlyText\" [model]=\"model.locReadOnlyText\"></sv-ng-string>\n        </div>\n      </div>\n        <sv-ng-popup [popupModel]=\"model.popupModel\"></sv-ng-popup>\n    </ng-container>\n    <div disabled *ngIf=\"model.isReadOnly\" [class]=\"model.getControlClass()\" [attr.id]=\"model.inputId\">\n      <sv-ng-string *ngIf=\"model.locReadOnlyText\" [model]=\"model.locReadOnlyText\"></sv-ng-string>\n    </div>\n  </div>\n</ng-template>\n", styles: [":host { display: none; }"], components: [{ type: i1.SurveyStringComponent, selector: "sv-ng-string, '[sv-ng-string]'", inputs: ["model"] }, { type: i1.PopupComponent, selector: "sv-ng-popup, '[sv-ng-popup]'", inputs: ["popupModel"] }], directives: [{ type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: LogicOperatorComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "svc-logic-operator",
                    templateUrl: "./logic-operator.component.html",
                    styles: [":host { display: none; }"]
                }]
        }] });
AngularComponentFactory.Instance.registerComponent("sv-logic-operator", LogicOperatorComponent);
RendererFactory.Instance.registerRenderer("dropdown", "logicoperator", "sv-logic-operator");
