// survey-creator-angular/esm2015/tabs/json/json-editor-textarea.component.js
import { Component, Input, ViewChild } from "@angular/core";
import { AngularComponentFactory, BaseAngular } from "survey-angular-ui";
import * as i0 from "@angular/core";
import * as i1 from "survey-angular-ui";
import * as i2 from "@angular/forms";
export class TextareaJsonEditorComponent extends BaseAngular {
    getModel() {
        return this.model;
    }
    ngOnInit() {
        this.model.canShowErrors = false;
        super.ngOnInit();
    }
    ngAfterViewInit() {
        this.model.textElement = this.inputEl.nativeElement;
    }
}
TextareaJsonEditorComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: TextareaJsonEditorComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
TextareaJsonEditorComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: TextareaJsonEditorComponent, selector: "svc-tab-json-editor-textarea", inputs: { model: "model" }, viewQueries: [{ propertyName: "inputEl", first: true, predicate: ["inputEl"], descendants: true }], usesInheritance: true, ngImport: i0, template: "<ng-template #template>\n  <div class=\"svc-creator-tab__content\">\n    <div class=\"svc-json-editor-tab__content\">\n      <textarea class=\"svc-json-editor-tab__content-area\" [attr.aria-label]=\"model.ariaLabel\"\n        [disabled]=\"model.readOnly\" [(ngModel)]=\"model.text\" (keydown)=\"model.checkKey(model, $event)\" #inputEl>\n      </textarea>\n      <div class=\"svc-json-editor-tab__errros_list\" [visible]=\"model.hasErrors\">\n        <sv-ng-list [model]=\"model.errorList\"></sv-ng-list>\n      </div>\n    </div>\n  </div>\n</ng-template>", styles: [":host { display: none; }"], components: [{ type: i1.ListComponent, selector: "sv-ng-list, '[sv-ng-list]'", inputs: ["model"] }], directives: [{ type: i2.DefaultValueAccessor, selector: "input:not([type=checkbox])[formControlName],textarea[formControlName],input:not([type=checkbox])[formControl],textarea[formControl],input:not([type=checkbox])[ngModel],textarea[ngModel],[ngDefaultControl]" }, { type: i2.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { type: i2.NgModel, selector: "[ngModel]:not([formControlName]):not([formControl])", inputs: ["name", "disabled", "ngModel", "ngModelOptions"], outputs: ["ngModelChange"], exportAs: ["ngModel"] }, { type: i1.VisibleDirective, selector: "[visible]", inputs: ["visible"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: TextareaJsonEditorComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "svc-tab-json-editor-textarea",
                    templateUrl: "./json-editor-textarea.component.html",
                    styles: [":host { display: none; }"]
                }]
        }], propDecorators: { model: [{
                type: Input
            }], inputEl: [{
                type: ViewChild,
                args: ["inputEl"]
            }] } });
AngularComponentFactory.Instance.registerComponent("svc-tab-json-editor-textarea", TextareaJsonEditorComponent);
