// survey-creator-angular/esm2015/side-bar/side-bar-header.component.js
import { Component, Input } from "@angular/core";
import { AngularComponentFactory, BaseAngular } from "survey-angular-ui";
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
export class SidebarHeaderComponent extends BaseAngular {
    getModel() {
        return this.model;
    }
}
SidebarHeaderComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: SidebarHeaderComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
SidebarHeaderComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: SidebarHeaderComponent, selector: "svc-side-bar-header", inputs: { model: "model" }, usesInheritance: true, ngImport: i0, template: "<ng-template #template>\n  <div class=\"svc-side-bar__container-header svc-sidebar__header-container\">\n    <ng-container *ngIf=\"!model.subTitle\">\n      <div class=\"svc-side-bar__container-title\">{{model.title}}</div>\n    </ng-container>\n    <ng-container *ngIf=\"!!model.subTitle\">\n      <div class=\"svc-sidebar__header-caption\">\n        <span class=\"svc-sidebar__header-title\">{{model.title}}</span>\n        <span class=\"svc-sidebar__header-subtitle\">{{model.subTitle}}</span>\n      </div>\n    </ng-container>\n  </div>\n</ng-template>", styles: [":host { display: none; }"], directives: [{ type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: SidebarHeaderComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "svc-side-bar-header",
                    templateUrl: "./side-bar-header.component.html",
                    styles: [":host { display: none; }"]
                }]
        }], propDecorators: { model: [{
                type: Input
            }] } });
AngularComponentFactory.Instance.registerComponent("svc-side-bar-header", SidebarHeaderComponent);
