// survey-creator-angular/esm2015/side-bar/tabs.component.js
import { Component, Input } from "@angular/core";
import { BaseAngular } from "survey-angular-ui";
import * as i0 from "@angular/core";
import * as i1 from "./tab-button.component";
import * as i2 from "@angular/common";
export class TabsComponent extends BaseAngular {
    getModel() {
        return this.model;
    }
}
TabsComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: TabsComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
TabsComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: TabsComponent, selector: "svc-tabs", inputs: { model: "model" }, usesInheritance: true, ngImport: i0, template: "<ng-template #template>\n  <ng-container *ngFor=\"let action of model.actions\">\n    <svc-tab-button [model]=\"action\"></svc-tab-button>\n  </ng-container>\n</ng-template>", styles: [":host { display: none; }"], components: [{ type: i1.TabButtonComponent, selector: "svc-tab-button", inputs: ["model"] }], directives: [{ type: i2.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: TabsComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "svc-tabs",
                    templateUrl: "./tabs.component.html",
                    styles: [":host { display: none; }"]
                }]
        }], propDecorators: { model: [{
                type: Input
            }] } });
