// survey-angular-ui/esm2015/components/breadcrumbs/breadcrumbs.component.js
import { Component, Input } from "@angular/core";
import { BaseAngular } from "../../base-angular";
import * as i0 from "@angular/core";
import * as i1 from "../svg-icon/svg-icon.component";
import * as i2 from "../action-bar/action.component";
import * as i3 from "@angular/common";
export class BreadcrumbsComponent extends BaseAngular {
    getModel() {
        return this.model;
    }
}
BreadcrumbsComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: BreadcrumbsComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
BreadcrumbsComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: BreadcrumbsComponent, selector: "sv-breadcrumbs", inputs: { model: "model", css: "css" }, usesInheritance: true, ngImport: i0, template: "<div *ngIf=\"model.actions && model.actions.length\" [class]=\"css.breadcrumbsRoot\">\n  <ng-container *ngFor=\"let item of model.actions; let itemIndex = index\">\n    <svg\n      *ngIf=\"itemIndex\"\n      [class]=\"css.breadcrumbsSeparator\"\n      [iconName]=\"'arrowright-16x16'\"\n      [size]=\"'auto'\"\n      sv-ng-svg-icon\n    ></svg>\n    <sv-ng-action [model]=\"item\"></sv-ng-action>\n  </ng-container>\n</div> ", components: [{ type: i1.SvgIconComponent, selector: "'[sv-ng-svg-icon]'", inputs: ["size", "width", "height", "iconName", "partCss", "css", "title"] }, { type: i2.ActionComponent, selector: "sv-ng-action", inputs: ["model"] }], directives: [{ type: i3.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i3.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: BreadcrumbsComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "sv-breadcrumbs",
                    templateUrl: "./breadcrumbs.component.html",
                    //styles: [":host { display: none; }"],
                }]
        }], propDecorators: { model: [{
                type: Input
            }], css: [{
                type: Input
            }] } });
