// survey-creator-angular/esm2015/tabbed-menu/tabbed-menu/tabbed-menu.component.js
import { Component, Input, ViewChild } from "@angular/core";
import { AngularComponentFactory, BaseAngular } from "survey-angular-ui";
import * as i0 from "@angular/core";
import * as i1 from "./tabbed-menu-item-wrapper.component";
import * as i2 from "@angular/common";
export class TabbledMenuComponent extends BaseAngular {
    getModel() {
        return this.model;
    }
    initResponsivityManager() {
        var _a;
        if ((_a = this.container) === null || _a === void 0 ? void 0 : _a.nativeElement) {
            this.model.initResponsivityManager(this.container.nativeElement);
        }
    }
    ngAfterViewInit() {
        this.initResponsivityManager();
    }
    afterUpdate(isSync) {
        super.afterUpdate(isSync);
        this.initResponsivityManager();
    }
    ngAfterViewChecked() {
        super.ngAfterViewChecked();
        this.initResponsivityManager();
    }
    ngOnDestroy() {
        super.ngOnDestroy();
        this.model.resetResponsivityManager();
    }
    trackItemBy(_, item) {
        return item.renderedId;
    }
}
TabbledMenuComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: TabbledMenuComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
TabbledMenuComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: TabbledMenuComponent, selector: "svc-tabbed-menu", inputs: { model: "model" }, viewQueries: [{ propertyName: "container", first: true, predicate: ["container"], descendants: true }], usesInheritance: true, ngImport: i0, template: "<ng-template #template>\n  <div class=\"svc-tabbed-menu\" #container role=\"tablist\" [style]=\"model.getRootStyle()\">\n    <ng-container *ngFor=\"let action of model.renderedActions; trackBy: trackItemBy\">\n      <svc-tabbed-menu-item-wrapper [model]=\"action\"></svc-tabbed-menu-item-wrapper>\n    </ng-container>\n  </div>\n</ng-template>", styles: [":host { display: none; }"], components: [{ type: i1.TabbedMenuItemWrapperComponent, selector: "svc-tabbed-menu-item-wrapper", inputs: ["model"] }], directives: [{ type: i2.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: TabbledMenuComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "svc-tabbed-menu",
                    templateUrl: "./tabbed-menu.component.html",
                    styles: [":host { display: none; }"]
                }]
        }], propDecorators: { model: [{
                type: Input
            }], container: [{
                type: ViewChild,
                args: ["container"]
            }] } });
AngularComponentFactory.Instance.registerComponent("svc-tabbed-menu", TabbledMenuComponent);
