// survey-angular-ui/esm2015/components/slider/slider-label-item.component.js
import { Component, Input } from "@angular/core";
import { BaseAngular } from "../../base-angular";
import { AngularComponentFactory } from "../../component-factory";
import * as i0 from "@angular/core";
import * as i1 from "../../survey-string.component";
import * as i2 from "@angular/common";
export class SliderLabelItemComponent extends BaseAngular {
    // onClick(event: any): void {
    //   this.model.setValueFromClick(event.target.value);
    //   event.stopPropagation();
    // }
    getModel() {
        return this.item;
    }
}
SliderLabelItemComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: SliderLabelItemComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
SliderLabelItemComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: SliderLabelItemComponent, selector: "sv-ng-slider-label-item", inputs: { model: "model", item: "item" }, usesInheritance: true, ngImport: i0, template: "<ng-template #template>\n    <div [class]=\"model.getLabelCss(item.locText)\"\n      [style]=\"{ left: model.getPercent(item.value) + '%' }\"\n      (pointerup)=\"model.handleLabelPointerUp($any($event), item.value)\">\n        <div [class]=\"model.cssClasses.labelTick\"></div>\n        <div [class]=\"model.cssClasses.labelTextContainer\">\n          <div *ngIf=\"!item.showValue\" [class]=\"model.cssClasses.labelText\" [model]=\"item.locText\" sv-ng-string></div>\n          <div *ngIf=\"item.showValue\" [class]=\"model.cssClasses.labelText\">{{ item.value }}</div>\n          <div *ngIf=\"item.showValue\" [class]=\"model.cssClasses.labelTextSecondaryMode\" [model]=\"item.locText\" sv-ng-string></div>\n        </div>\n    </div>\n</ng-template>", components: [{ type: i1.SurveyStringComponent, selector: "sv-ng-string, '[sv-ng-string]'", inputs: ["model"] }], directives: [{ type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: SliderLabelItemComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "sv-ng-slider-label-item",
                    templateUrl: "./slider-label-item.component.html"
                }]
        }], propDecorators: { model: [{
                type: Input
            }], item: [{
                type: Input
            }] } });
AngularComponentFactory.Instance.registerComponent("sv-slider-label-item", SliderLabelItemComponent);
