// survey-creator-angular/esm2015/tabs/json/json-error-item.component.js
import { Component, Input } from "@angular/core";
import { AngularComponentFactory, BaseAngular } from "survey-angular-ui";
import * as i0 from "@angular/core";
import * as i1 from "survey-angular-ui";
import * as i2 from "@angular/common";
export class JsonErrorItemComponent extends BaseAngular {
    getModel() {
        return this.model;
    }
    fixError(event) {
        event.stopPropagation();
        this.model.data.fixError();
    }
}
JsonErrorItemComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: JsonErrorItemComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
JsonErrorItemComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: JsonErrorItemComponent, selector: "svc-json-error-item", inputs: { model: "model" }, usesInheritance: true, ngImport: i0, template: "<ng-template #template>\n    <svg [iconName]=\"model.iconName\" [size]=\"model.iconSize\" [css]=\"'svc-json-error__icon'\" sv-ng-svg-icon></svg>\n    <div class=\"svc-json-error__container\">\n        <div class=\"svc-json-error__title\">\n            <sv-ng-string [model]=\"model.locTitle\"></sv-ng-string>\n        </div>\n        <button type=\"button\" *ngIf=\"model.data.showFixButton\" (click)=\"fixError($event)\" [key2click]\n            [attr.title]=\"model.data.fixButtonTitle\" [attr.aria-label]=\"model.data.fixButtonTitle\"\n            class=\"svc-json-error__fix-button\">\n            <svg [iconName]=\"model.data.fixButtonIcon\" [size]=\"'auto'\" sv-ng-svg-icon></svg>\n        </button>\n    </div>\n</ng-template>", styles: [":host { display: none }"], components: [{ type: i1.SvgIconComponent, selector: "'[sv-ng-svg-icon]'", inputs: ["size", "width", "height", "iconName", "partCss", "css", "title"] }, { type: i1.SurveyStringComponent, selector: "sv-ng-string, '[sv-ng-string]'", inputs: ["model"] }], directives: [{ type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i1.Key2ClickDirective, selector: "[key2click]", inputs: ["key2click"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: JsonErrorItemComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "svc-json-error-item",
                    templateUrl: "./json-error-item.component.html",
                    styles: [":host { display: none }"]
                }]
        }], propDecorators: { model: [{
                type: Input
            }] } });
AngularComponentFactory.Instance.registerComponent("json-error-item", JsonErrorItemComponent);
