// survey-creator-angular/esm2015/tabs/json/json-editor-ace.component.js
import { Component, Input, ViewChild } from "@angular/core";
import { AngularComponentFactory, BaseAngular } from "survey-angular-ui";
import * as i0 from "@angular/core";
import * as i1 from "survey-angular-ui";
export class AceJsonEditorComponent extends BaseAngular {
    getModel() {
        return this.model;
    }
    ngAfterViewInit() {
        //todo fix ts 
        this.model.init(window.ace.edit(this.inputEl.nativeElement));
    }
}
AceJsonEditorComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: AceJsonEditorComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
AceJsonEditorComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: AceJsonEditorComponent, selector: "svc-tab-json-editor-ace", inputs: { model: "model" }, viewQueries: [{ propertyName: "inputEl", first: true, predicate: ["inputEl"], descendants: true }], usesInheritance: true, ngImport: i0, template: "<ng-template #template>\n    <div class=\"svc-creator-tab__content\">\n        <div class=\"svc-json-editor-tab__content\">\n            <div class=\"svc-json-editor-tab__ace-editor\" #inputEl></div>\n            <div class=\"svc-json-editor-tab__errros_list\" [visible]=\"model.hasErrors\">\n                <sv-ng-list [model]=\"model.errorList\"></sv-ng-list>\n            </div>\n        </div>\n    </div>\n</ng-template>", styles: [":host { display: none; }"], components: [{ type: i1.ListComponent, selector: "sv-ng-list, '[sv-ng-list]'", inputs: ["model"] }], directives: [{ type: i1.VisibleDirective, selector: "[visible]", inputs: ["visible"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: AceJsonEditorComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "svc-tab-json-editor-ace",
                    templateUrl: "./json-editor-ace.component.html",
                    styles: [":host { display: none; }"]
                }]
        }], propDecorators: { model: [{
                type: Input
            }], inputEl: [{
                type: ViewChild,
                args: ["inputEl"]
            }] } });
AngularComponentFactory.Instance.registerComponent("svc-tab-json-editor-ace", AceJsonEditorComponent);
