// survey-creator-angular/esm2015/side-bar/property-grid.component.js
import { Component, Input } from "@angular/core";
import { AngularComponentFactory, BaseAngular, ButtonGroupQuestionComponent } from "survey-angular-ui";
import * as i0 from "@angular/core";
import * as i1 from "../components/search.component";
import * as i2 from "../components/scroll.component";
import * as i3 from "survey-angular-ui";
import * as i4 from "@angular/common";
export class PropertyGridComponent extends BaseAngular {
    getModel() {
        return this.model;
    }
    getPropertiesToUpdateSync() {
        return ["survey"];
    }
}
PropertyGridComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: PropertyGridComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
PropertyGridComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: PropertyGridComponent, selector: "svc-property-grid", inputs: { model: "model" }, usesInheritance: true, ngImport: i0, template: "<ng-template #template>\n    <div [class]=\"model.rootCss\">\n        <svc-search [model]=\"model.searchManager\"></svc-search>\n        <svc-scroll>\n            <survey-content *ngIf=\"!!model.survey\" [model]=\"model.survey\"></survey-content>\n        </svc-scroll>\n    </div>\n</ng-template>", styles: [":host { display: none; }"], components: [{ type: i1.SearchComponent, selector: "svc-search", inputs: ["model"] }, { type: i2.ScrollComponent, selector: "svc-scroll" }, { type: i3.SurveyContentComponent, selector: "survey-content", inputs: ["model"] }], directives: [{ type: i4.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: PropertyGridComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "svc-property-grid",
                    templateUrl: "./property-grid.component.html",
                    styles: [":host { display: none; }"]
                }]
        }], propDecorators: { model: [{
                type: Input
            }] } });
AngularComponentFactory.Instance.registerComponent("buttongroup-question", ButtonGroupQuestionComponent);
AngularComponentFactory.Instance.registerComponent("svc-property-grid", PropertyGridComponent);
