// survey-creator-angular/esm2015/side-bar/side-bar.component.js
import { Component, Input, ViewChild } from "@angular/core";
import { BaseAngular } from "survey-angular-ui";
import * as i0 from "@angular/core";
import * as i1 from "./side-bar-page.component";
import * as i2 from "survey-angular-ui";
import * as i3 from "@angular/common";
export class SidebarComponent extends BaseAngular {
    getModel() {
        return this.model;
    }
    ngOnDestroy() {
        this.model.resetResizeManager();
        super.ngOnDestroy();
    }
    ngAfterViewInit() {
        this.model.initResizeManager(this.container.nativeElement);
    }
}
SidebarComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: SidebarComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
SidebarComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: SidebarComponent, selector: "svc-side-bar", inputs: { model: "model" }, viewQueries: [{ propertyName: "container", first: true, predicate: ["container"], descendants: true }], usesInheritance: true, ngImport: i0, template: "<ng-template #template>\n  <div [class]=\"model.rootCss\" [visible]=\"model.renderRoot\">\n    <div class=\"svc-side-bar__shadow\" (click)=\"model.collapseSidebar()\" [visible]=\"model.renderContainer\"></div>\n    <div class=\"svc-flex-row svc-side-bar__wrapper\">\n      <div class=\"svc-side-bar__container-wrapper\" [visible]=\"model.renderContainer\">\n        <div class=\"svc-side-bar__container\" [visible]=\"model.renderedIsVisible\" #container>\n          <ng-template\n            [component]=\"{ name: model.header.component, data: { model: model.header.componentModel } }\"></ng-template>\n          <div class=\"svc-side-bar__container-content\">\n            <ng-container *ngFor=\"let page of model.pages\">\n              <svc-side-bar-page [model]=\"page\"></svc-side-bar-page>\n            </ng-container>\n          </div>\n        </div>\n      </div>\n      <ng-container *ngIf=\"model.sideAreaComponentName\">\n        <ng-template\n          [component]=\"{ name: model.sideAreaComponentName, data: { model: model.sideAreaComponentData } }\"></ng-template>\n      </ng-container>\n    </div>\n  </div>\n</ng-template>", styles: [":host { display: none; }"], components: [{ type: i1.SidebarPageComponent, selector: "svc-side-bar-page", inputs: ["model"] }], directives: [{ type: i2.VisibleDirective, selector: "[visible]", inputs: ["visible"] }, { type: i2.DynamicComponentDirective, selector: "[component]", inputs: ["component"] }, { type: i3.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { type: i3.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: SidebarComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "svc-side-bar",
                    templateUrl: "./side-bar.component.html",
                    styles: [":host { display: none; }"]
                }]
        }], propDecorators: { model: [{
                type: Input
            }], container: [{
                type: ViewChild,
                args: ["container"]
            }] } });
