// survey-creator-angular/esm2015/toolbox/adaptive-toolbox.component.js
import { Component, Input, ViewChild } from "@angular/core";
import { AngularComponentFactory, BaseAngular } from "survey-angular-ui";
import * as i0 from "@angular/core";
import * as i1 from "./toolbox-tool.component";
import * as i2 from "../components/search.component";
import * as i3 from "survey-angular-ui";
import * as i4 from "@angular/common";
export class AdaptiveToolboxComponent extends BaseAngular {
    get toolbox() {
        return this.model.toolbox;
    }
    get creator() {
        return this.model;
    }
    get searchItem() {
        return this.toolbox.searchItem;
    }
    afterRender() {
        var _a;
        if ((_a = this.container) === null || _a === void 0 ? void 0 : _a.nativeElement) {
            this.toolbox.afterRender(this.container.nativeElement);
        }
    }
    ngAfterViewInit() {
        this.afterRender();
    }
    ngAfterViewChecked() {
        super.ngAfterViewChecked();
        this.afterRender();
    }
    afterUpdate(isSync) {
        super.afterUpdate(isSync);
        this.afterRender();
    }
    getModel() {
        return this.toolbox;
    }
    ngOnDestroy() {
        this.toolbox.beforeDestroy();
        super.ngOnDestroy();
    }
    trackItemBy(_, item) {
        return item.renderedId;
    }
}
AdaptiveToolboxComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: AdaptiveToolboxComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
AdaptiveToolboxComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: AdaptiveToolboxComponent, selector: "svc-toolbox", inputs: { model: "model" }, viewQueries: [{ propertyName: "container", first: true, predicate: ["container"], descendants: true }], usesInheritance: true, ngImport: i0, template: "<ng-template #template>\n  <div [class]=\"toolbox.classNames\" #container [style]=\"toolbox.getRootStyle()\">\n    <div (focusout)=\"toolbox.focusOut($event)\" class=\"svc-toolbox__panel\">\n      <div *ngIf=\"toolbox.showSearch\" class=\"svc-toolbox__search-container\">\n        <ng-container *ngIf=\"toolbox.isCompactRendered\">\n          <svc-toolbox-tool [creator]=\"creator\" [item]=\"searchItem\" [parentModel]=\"toolbox\"\n            [isCompact]=\"toolbox.isCompactRendered\"></svc-toolbox-tool>\n        </ng-container>\n        <svc-search [model]=\"toolbox.searchManager\"></svc-search>\n        <div class=\"svc-toolbox__category-separator svc-toolbox__category-separator--search\"></div>\n      </div>\n      <div *ngIf=\"toolbox.showPlaceholder\" class=\"svc-toolbox__placeholder\">{{toolbox.toolboxNoResultsFound}}</div>\n      <sv-scroll>\n        <ng-container *ngIf=\"!toolbox.showInSingleCategory\">\n          <ng-container *ngFor=\"let category of toolbox.categories\">\n            <ng-template [component]=\"{name: 'svc-toolbox-category', data: { category: category, toolbox: toolbox }}\"></ng-template>\n          </ng-container>\n        </ng-container>\n        <ng-container *ngIf=\"toolbox.showInSingleCategory\">\n          <div class=\"svc-toolbox__category\">\n            <ng-container *ngFor=\"let item of toolbox.renderedActions;  trackBy: trackItemBy\">\n              <svc-toolbox-tool [creator]=\"creator\" [item]=\"item\" [parentModel]=\"toolbox\"\n                [isCompact]=\"toolbox.isCompactRendered\"></svc-toolbox-tool>\n            </ng-container>\n          </div>\n        </ng-container>\n      </sv-scroll>\n    </div>\n  </div>\n</ng-template>", styles: [":host { display: none; }"], components: [{ type: i1.ToolboxToolComponent, selector: "svc-toolbox-tool", inputs: ["creator", "item", "parentModel", "isCompact"] }, { type: i2.SearchComponent, selector: "svc-search", inputs: ["model"] }, { type: i3.ScrollComponent, selector: "sv-scroll", inputs: ["disabled", "onInnerHeightChanged"] }], directives: [{ type: i4.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i4.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { type: i3.DynamicComponentDirective, selector: "[component]", inputs: ["component"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: AdaptiveToolboxComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "svc-toolbox",
                    templateUrl: "./adaptive-toolbox.component.html",
                    styles: [":host { display: none; }"]
                }]
        }], propDecorators: { model: [{
                type: Input
            }], container: [{
                type: ViewChild,
                args: ["container"]
            }] } });
AngularComponentFactory.Instance.registerComponent("svc-toolbox", AdaptiveToolboxComponent);
