// survey-creator-angular/esm2015/toolbox/toolbox-tool.component.js
import { Component, Input, ViewChild } from "@angular/core";
import { CreatorModelComponent } from "../creator-model.component";
import { ToolboxToolViewModel } from "survey-creator-core";
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
import * as i2 from "survey-angular-ui";
export class ToolboxToolComponent extends CreatorModelComponent {
    constructor() {
        super(...arguments);
        this.isCompact = false;
    }
    createModel() {
        this.model = new ToolboxToolViewModel(this.item, this.creator, this.parentModel);
    }
    getPropertiesToTrack() {
        return ["creator", "item"];
    }
    getModel() {
        return this.item;
    }
    get itemCssClasses() {
        return this.item.css;
    }
    ngOnDestroy() {
        super.ngOnDestroy();
        this.model.dispose();
        this.item.updateModeCallback = undefined;
    }
    ngAfterViewInit() {
        if (this.item) {
            this.item.updateModeCallback = (mode, callback) => {
                var _a;
                this.item.mode = mode;
                callback(mode, (_a = this.container) === null || _a === void 0 ? void 0 : _a.nativeElement);
            };
            queueMicrotask(() => {
                this.item.afterRender();
            });
        }
    }
}
ToolboxToolComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: ToolboxToolComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
ToolboxToolComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: ToolboxToolComponent, selector: "svc-toolbox-tool", inputs: { creator: "creator", item: "item", parentModel: "parentModel", isCompact: "isCompact" }, viewQueries: [{ propertyName: "container", first: true, predicate: ["container"], descendants: true }], usesInheritance: true, ngImport: i0, template: "<ng-template #template>\n    <div [class]=\"itemCssClasses\" #container>\n        <div class=\"svc-toolbox__category-separator\" *ngIf=\"item.needSeparator && !creator.toolbox.showCategoryTitles\">\n        </div>\n        <div class=\"svc-toolbox__tool-content sv-action__content\" (pointerdown)=\"model.onPointerDown($event)\">\n            <ng-template [component]=\"{ name: model.itemComponent, default: 'svc-toolbox-item', \n            data: { model: item, viewModel: model, creator: creator, isCompact: isCompact } }\"></ng-template>\n        </div>\n    </div>\n</ng-template>", styles: [":host { display: none; }"], directives: [{ type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i2.DynamicComponentDirective, selector: "[component]", inputs: ["component"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: ToolboxToolComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "svc-toolbox-tool",
                    templateUrl: "./toolbox-tool.component.html",
                    styles: [":host { display: none; }"]
                }]
        }], propDecorators: { creator: [{
                type: Input
            }], item: [{
                type: Input
            }], parentModel: [{
                type: Input
            }], isCompact: [{
                type: Input
            }], container: [{
                type: ViewChild,
                args: ["container"]
            }] } });
