// survey-creator-angular/esm2015/toolbox/toolbox-item-group.component.js
import { Component, Input } from "@angular/core";
import { BaseAngular, AngularComponentFactory } from "survey-angular-ui";
import * as i0 from "@angular/core";
import * as i1 from "./toolbox-item.component";
import * as i2 from "survey-angular-ui";
export class ToolboxItemGroupComponent extends BaseAngular {
    constructor() {
        super(...arguments);
        this.isCompact = false;
    }
    getModel() {
        return this.viewModel;
    }
    get item() {
        return this.model;
    }
}
ToolboxItemGroupComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: ToolboxItemGroupComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
ToolboxItemGroupComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: ToolboxItemGroupComponent, selector: "svc-toolbox-item-group", inputs: { creator: "creator", model: "model", isCompact: "isCompact", viewModel: "viewModel" }, usesInheritance: true, ngImport: i0, template: "<ng-template #template>\n  <svc-toolbox-item [model]=\"item\" [viewModel]=\"viewModel\" [creator]=\"creator\" [isCompact]=\"isCompact\"></svc-toolbox-item>\n  <div class=\"svc-toolbox__item-submenu-button\" (mouseover)=\"viewModel.onMouseOver(item, $event)\"\n    (mouseleave)=\"viewModel.onMouseLeave(item, $event)\">\n    <svg [iconName]=\"item.subitemsButtonIcon\" [size]=\"'auto'\" sv-ng-svg-icon></svg>\n    <sv-ng-popup [popupModel]=\"model.popupModel\"></sv-ng-popup>\n  </div>\n</ng-template>", styles: [":host { display: none; }"], components: [{ type: i1.ToolboxItemComponent, selector: "svc-toolbox-item", inputs: ["creator", "model", "isCompact", "viewModel"] }, { type: i2.SvgIconComponent, selector: "'[sv-ng-svg-icon]'", inputs: ["size", "width", "height", "iconName", "partCss", "css", "title"] }, { type: i2.PopupComponent, selector: "sv-ng-popup, '[sv-ng-popup]'", inputs: ["popupModel"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: ToolboxItemGroupComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "svc-toolbox-item-group",
                    templateUrl: "./toolbox-item-group.component.html",
                    styles: [":host { display: none; }"]
                }]
        }], propDecorators: { creator: [{
                type: Input
            }], model: [{
                type: Input
            }], isCompact: [{
                type: Input
            }], viewModel: [{
                type: Input
            }] } });
AngularComponentFactory.Instance.registerComponent("svc-toolbox-item-group", ToolboxItemGroupComponent);
